http/client.go

```diff
@@ -14,30 +14,45 @@ import (
 	"time"
 )
 
-// New creates a new http.Client with a custom certificate, which can be loaded from the passed CA Bundle file and/or
+type TransportOption func(*http.Transport) error
+
+// New creates a new http.Client with the Passed Transport Options
+// that will have a custom certificate if it is loaded from the passed CA Bundle file and/or
 // directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
-func New(CABundleFile, CABundleDir string, httpTimeout time.Duration) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, "")
-}
+func New(httpTimeout time.Duration, opts ...TransportOption) (*http.Client, error) {
```

Comment:
Feels a bit strange to have the Options pattern but `httpTimeout` as a different parameter.

Unfortunately, timeout is not part of `http.Transport`... So we would need to change that from

`type TransportOption func(*http.Transport) error`
to
`type ClientOption func(*http.Client) error`

Which I'm not sure it's something we want. What are your thoughts? (also @varas)

Reply:
Since Timeout and Transport are public and can be modified after creation, I can definitelly do as you say, type ClientOption func(*http.Client), creating first of all the http.Client and modifying the timeout or the transport values in each configOption, do you agree?